lisa/base_tools/wget.py

```diff
@@ -47,6 +47,10 @@ def get(
     ) -> str:
         is_valid_url(url)
 
+        if not filename:
```

Comment:
When is it needed? The `wget` should give a name by default. The web downloader obtains the name from a specific header firstly. If it doesn't specified, then use the last part of url. But below logic doesn't consider the header and it includes all parts after last `/`, and it may include the query string, which shouldn't be a part of the file name.

The following logic parses the file name from download log. Please check what's wrong with it, and propose a fix.

Reply:
The issue is that Wget.get does not cache results the same way that Tool.run does. Maybe we can add caching. Just so you're aware, that same filename default is already present in the Windows version of the tool. I had hoped that since it was already present it wouldn't be an issue to add it to Linux too. We can fix both.

The bug is this:

Without force_run, Wget.get will:
1. check if the directory exists. (but not check the filename arg)
2. remove the directory if overwrite is present.
1. download the file with Wget.run(... force_run=force_run)
3. check the log file for the filename if the log file exists
4. deletes the log file.
5. uses LS to try to find the filename if the log file isn't present. It uses the filename and path arugments in this case. But both are optional.
6. returns the filename

All of those steps happen every time, even if force_run is False. If force_run is False, the log file is not available since it's deleted. So if filename is not present and filepath *is* present (the default case fills it in as the working path), the tool will return the entire contents of the directory as the filename. So LISA ends up trying to use .../working_path/dpdk\trdma-core\t as a tar file.

Reply:
> But below logic doesn't consider the header and it includes all parts after last `/`, and it may include the query string, which shouldn't be a part of the file name.

This also isn't entirely accurate. urlparse seperates the path and query. An example would be urlparse(url).path will return just `/path/to/file.tar.gz` from `https://[place]/path/to/file.tar.gz?query_string=query`

Reply:
I see, please cache a map with (url, returned file name). If force_run is false, check the dict firstly.

Reply:
Thanks! Added